### PR TITLE
New package: Icinga2

### PR DESCRIPTION
**Pull request by EaterOfCode**
This will add the icinga2 package to void repository.

### CODE REVIEW

File: srcpkgs/monitoring-plugins/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+maintainer="Corné Oppelaar <hello@eaterofco.de>"
+license="GPL-3"
+homepage="https://www.monitoring-plugins.org/"
+distfiles="https://www.monitoring-plugins.org/download/monitoring-plugins-${version}.tar.gz"
+checksum="76c6b58f0867ab7b6c8c86c7e94fcce7183618f6daab63488990d0355f5600aa"
+
+do_configure() {
+	: ${configure_script:=./configure}
+
+	${configure_script} ${configure_args} --with-ping-command='/usr/bin/iputils-ping -n -U -w %d -c %d %s' --with-ping6-command='/usr/bin/iputils-ping6 -n -U -w %d -c %d %s'
```
Comment (Vaelatern) [inline srcpkgs/monitoring-plugins/template]: If @jaagr is correct on this, you can `configure_args=--libexecdir=/usr/lib/monitoring-plugins\ --with-ping-command="/usr/bin/iputils-ping -n -U -w %d -c %d %s"\ --with-ping6-command="/usr/bin/iputils-ping6 -n -U -w %d -c %d %s"`

**Comment by the-maldridge** [comment]
This package satisfies dependency "nagios-nrpe-plugin" on #3119.

**Comment by EaterOfCode** [comment]
@the-maldridge

This should also satisfy "nagios-plugins"

File: srcpkgs/monitoring-plugins/template (hunk 1 above)
Comment (EaterOfCode) [inline srcpkgs/monitoring-plugins/template]: this results in my `./xbps-src pkg monitoring-plugins` failing with:

```
=> Using `/home/eater/projects/void-packages/hostdir/binpkgs/addMonitoringPlugins' as local repository.
[*] Updating `http://repo.voidlinux.eu/current/x86_64-repodata' ...
[*] Updating `http://repo.voidlinux.eu/current/nonfree/x86_64-repodata' ...
[*] Updating `http://muslrepo.voidlinux.eu/current/x86_64-repodata' ...
ERROR: [reposync] failed to fetch file `http://muslrepo.voidlinux.eu/current/x86_64-repodata': Not Found
[*] Updating `http://muslrepo.voidlinux.eu/current/nonfree/x86_64-repodata' ...
ERROR: [reposync] failed to fetch file `http://muslrepo.voidlinux.eu/current/nonfree/x86_64-repodata': Not Found
[*] Updating `http://repo.voidlinux.eu/current/multilib/x86_64-repodata' ...
[*] Updating `http://repo.voidlinux.eu/current/multilib/nonfree/x86_64-repodata' ...
=> monitoring-plugins-2.1.2_1: removing autodeps, please wait...
=> monitoring-plugins-2.1.2_1: building ...
   [target] libressl-devel-2.2.6_1: found (http://repo.voidlinux.eu/current)
   [target] libldap-devel-2.4.44_1: found (http://repo.voidlinux.eu/current)
   [target] fping-3.13_1: found (http://repo.voidlinux.eu/current)
   [target] postgresql-libs-devel-9.5.1_2: found (http://repo.voidlinux.eu/current)
   [target] libmysqlclient-5.6.29_1: found (http://repo.voidlinux.eu/current)
   [target] libmysqlclient-devel-5.6.29_1: found (http://repo.voidlinux.eu/current)
   [target] procps-ng-3.3.11_4: found (http://repo.voidlinux.eu/current)
   [target] zlib-devel-1.2.8_6: found (http://repo.voidlinux.eu/current)
   [runtime] iputils>=0: found (http://repo.voidlinux.eu/current)
   [runtime] procps-ng>=0: found (http://repo.voidlinux.eu/current)
=> monitoring-plugins-2.1.2_1: installing target dependency 'libressl-devel-2.2.6_1' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'libldap-devel-2.4.44_1' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'fping-3.13_1' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'postgresql-libs-devel-9.5.1_2' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'libmysqlclient-5.6.29_1' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'libmysqlclient-devel-5.6.29_1' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'procps-ng-3.3.11_4' ...
=> monitoring-plugins-2.1.2_1: installing target dependency 'zlib-devel-1.2.8_6' ...
=> monitoring-plugins-2.1.2_1: running do-fetch hook: 00-distfiles ...
=> monitoring-plugins-2.1.2_1: running do-extract hook: 00-distfiles ...
=> monitoring-plugins-2.1.2_1: extracting distfile(s), please wait...
=> monitoring-plugins-2.1.2_1: running post-extract hook: 00-patches ...
=> monitoring-plugins-2.1.2_1: running pre-configure hook: 00-gnu-configure-asneeded ...
=> monitoring-plugins-2.1.2_1: running pre-configure hook: 01-override-config ...
=> monitoring-plugins-2.1.2_1: running pre-configure hook: 02-script-wrapper ...
=> monitoring-plugins-2.1.2_1: running do_configure ...
configure: error: unrecognized option: `-U'
Try `./configure --help' for more information
=> ERROR: monitoring-plugins-2.1.2_1: failed to run do_configure() at line 7.
```

**Comment by EaterOfCode** [comment]
Currently only the builds fail because they run out of memory :(

File: srcpkgs/monitoring-plugins/template (hunk 1 above)
Comment (Vaelatern) [inline srcpkgs/monitoring-plugins/template]: I have not been able to get this to work with any configuration, not without providing the functions you did here.

Comment (EaterOfCode) [inline srcpkgs/monitoring-plugins/template]: So my proposal is correct in this case?

Comment (Vaelatern) [inline srcpkgs/monitoring-plugins/template]: Yes, to the best of my testing.

**Comment by EaterOfCode** [comment]
@Vaelatern any other problems in this merge?

File: srcpkgs/icinga2/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+makedepends="libressl-devel boost-devel yajl-devel libmysqlclient-devel libedit-devel postgresql-libs-devel"
+depends="monitoring-plugins"
+maintainer="Corné Oppelaar <hello@eaterofco.de>"
+license="GPL-2"
+homepage="https://www.icinga.org/"
+distfiles="https://github.com/Icinga/icinga2/archive/v${version}.tar.gz"
+checksum="ce19ed52c447929f5f9dae11b75d080b76b7aa680ee930c59db4202a2218d6b5"
+
+system_groups="icingacmd icinga"
+system_accounts="icinga"
```
Comment (Vaelatern) [inline srcpkgs/icinga2/template]: Change this order.
```
system_accounts="icinga"
system_groups="icingacmd"
```

File: srcpkgs/icinga2/template [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'icinga2'
+pkgname="icinga2"
+version="2.4.4"
+revision=1
+short_desc="Icinga is a scalable and extensible monitoring system"
+nocross="yes"
+build_style="cmake"
+configure_args="-DCMAKE_INSTALL_SYSCONFDIR=/etc -DICINGA2_RUNDIR=/run -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_PLUGINDIR=/usr/lib/monitoring-plugins/"
+hostmakedepends="bison flex git "
```
Comment (Vaelatern) [inline srcpkgs/icinga2/template]: `hostmakedepends="flex git"`

**Comment by EaterOfCode** [comment]
@Vaelatern processed feedback and edited the commit accordingly

**Comment by Vaelatern** [comment]
Fails on my machine,

```
[ 86%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesOptionsWidget.cxx.o
[ 87%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesBoolWidget.cxx.o
[ 87%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesCacheEntryComposite.cxx.o
[ 87%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesDummyWidget.cxx.o
[ 88%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesFilePathWidget.cxx.o
[ 88%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesForm.cxx.o
[ 89%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesLabelWidget.cxx.o
[ 89%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesLongMessageForm.cxx.o
[ 89%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesMainForm.cxx.o
[ 90%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesPathWidget.cxx.o
[ 90%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesStringWidget.cxx.o
[ 90%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/cmCursesWidget.cxx.o
[ 91%] Building CXX object Source/CMakeFiles/ccmake.dir/CursesDialog/ccmake.cxx.o
make[2]: *** No rule to make target '/usr/lib64/libarchive.so', needed by 'bin/ccmake'.  Stop.
CMakeFiles/Makefile2:1539: recipe for target 'Source/CMakeFiles/ccmake.dir/all' failed
make[1]: *** [Source/CMakeFiles/ccmake.dir/all] Error 2
Makefile:146: recipe for target 'all' failed
make: *** [all] Error 2
```

File: srcpkgs/icinga2/template [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'icinga2'
+pkgname="icinga2"
+version="2.4.4"
+revision=1
+short_desc="Icinga is a scalable and extensible monitoring system"
+nocross="yes"
+build_style="cmake"
+configure_args="-DCMAKE_INSTALL_SYSCONFDIR=/etc -DICINGA2_RUNDIR=/run -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_PLUGINDIR=/usr/lib/monitoring-plugins/"
+hostmakedepends="flex git "
+makedepends="libressl-devel boost-devel yajl-devel libmysqlclient-devel libedit-devel postgresql-libs-devel"
```
Comment (Vaelatern) [inline srcpkgs/icinga2/template]: Probably want to add `libarchive-devel`

**Comment by Vaelatern** [comment]
@EaterOfCode Any changes? The build definitely failed for me.

**Comment by EaterOfCode** [comment]
@Vaelatern I'm sorry I'm kinda busy lately, I will look into it soon! I'm sorry

**Comment by Vaelatern** [comment]
No problem, just making sure there is motion.

**Comment by Vaelatern** [comment]
If there is no motion, I'll adopt work on this.

**Comment by EaterOfCode** [comment]
Feel free to do so! Im very busy lately and couldn't find time to work on this

**Comment by the-maldridge** [comment]
Can we do a partial merge on this to go ahead and have monitoring-plugins?

**Pull request closed without merging**